proposals/NNNN-GetInteriorVehicleData.md

```diff
@@ -0,0 +1,96 @@
+# Interior Vehicle Data Subscription Management and Data Caching in SDL
+
+* Proposal: [SDL-NNNN](NNNN-GetInteriorVehicleData.md)
+* Author: [Zhimin Yang](https://github.com/smartdevicelink/yang1070)
+* Impacted Platforms: [Core / Web / RPC ]
+
+## Introduction
+
+This proposal is for improving the process of GetInteriorVehicleData RPC in SDL core. This proposal can reduce the number of messages between SDL and HMI, and improves the response time of GetInteriorVehicleData a lot. It accomplishes this by introducing a cache of the interior vehicle data and maintains SDL's subscription status of a module(Type). 
+
+
+## Motivation
+
+The following sequence diagram shows the current behavior of sdl regarding GetInteriorVehicleData.
+
+![../assets/proposals/nnnn-getInteriorVehicleData/GetInteriorVehicleData_current.png](../assets/proposals/nnnn-getInteriorVehicleData/GetInteriorVehicleData_current.png)
+
+#### 1.	SDL always forwards a valid GetInteriorVehicleData request from any app to HMI, no matter parameter `subscribe` included or not.
+#### 2. SDL expects HMI to manage each app’s subscription status (per moduleType) and maintains a local status keep tracking of each application’s subscription (per moduleType).
+#### 3. SDL mainly works as a pass-through/forwarder with little intelligence  
+- If there is no subscribe parameter in GetInteriorVehicleData, SDL forwards the request to HMI.
+- If parameter `subscribe=true` and the app has not subscribed to the module yet, SDL forward the whole request to HMI.
+- If parameter `subscribe=true` and the app has already subscribed to the module, SDL cut off `subscribe=true` from the request and forward it to HMI.
+- If parameter `subscribe=false` and the app has not subscribed to the module yet, SDL cut off  `subscribe=false` from the request and forward it to HMI.
+- If parameter `subscribe=false` and the app has already subscribed to the module, SDL forward the whole request to HMI.
+
+### Advantages:
+-	A simple SDL. Just forwarding the GetInteriorVehicleData requests and response between the app and the HMI.
+
+### Shortcomings:
+-	The operation of GetInteriorVehicleData and subscribe InteriorVehicleData on HMI side may be a time consuming task. 
+-	If app1 has already subscribed data change notification for a moduleType, SDL still need to include parameter `subscribe=true` when forwarding requests from other apps, i.e. to re-subscribe the data again. There are extra messages and extra processing time.
+-	If app1 has not subscribed data change notification for a moduleType, SDL still need to include parameter `subscribe=false` when forwarding requests from other apps, i.e. to un-subscribe the data per app. There are extra messages and extra processing time.
```

Comment:
Please update to "...SDL still needs to include..."

Reply:
updated